File: Wielermanager/leaderboard.py
# File: leaderboard.py
class Leaderboard:
    def __init__(self):
        self.scores = {}  # speler_naam: score (int)
        self.teams = {}   # speler_naam: FantasyTeam object

    def add_team(self, team):
        self.teams[team.player_name] = team

    def calculate_scores(self, employees):
        best = {
            'time_value': max(employees, key=lambda e: e.stats['time_value']).name,
            'coffee': max(employees, key=lambda e: e.stats['coffee']).name,
            'tires_changed': max(employees, key=lambda e: e.stats['tires_changed']).name,
            'returns': min(employees, key=lambda e: e.stats['returns']).name,
            'service_score': max(employees, key=lambda e: e.stats['service_score']).name,
            'lateness': max(employees, key=lambda e: e.stats['lateness']).name
        }

        for player_name, team in self.teams.items():
            score = sum(1 for cat, emp_name in team.picks.items() if best[cat] == emp_name)
            team.score = score
            self.scores[player_name] = score

    def show_leaderboard(self):
        sorted_scores = sorted(self.scores.items(), key=lambda x: x[1], reverse=True)
        print("=== Leaderboard ===")
        for player_name, score in sorted_scores:
            print(f"{player_name}: {score} punten")

File: Wielermanager/Front-End/leaderboard.js
async function loadLeaderboard() {
  try {
    const response = await fetch('./leaderboard_data.json');
    const data = await response.json();

    const list = document.getElementById('leaderboard-list');
    list.innerHTML = ''; // Oude lijst leegmaken

    // Sorteer spelers op score
    console.log("Passed first test")
    const sorted = Object.entries(data)
      .map(([name, info]) => ({
        name: name,
        score: info.score || 0
      }))
      .sort((a, b) => b.score - a.score);

    sorted.forEach(player => {
      const item = document.createElement('div');
      item.className = 'list-item';
      item.innerHTML = `
        <div class="list-item-content">
          <span class="item-text">${player.name}</span>
          <span class="item-score">${player.score}</span>
        </div>
      `;
      list.appendChild(item);
      console.log("Passed second test")
    });
  } catch (error) {
    console.error('❌ Fout bij laden leaderboard:', error);
    console.log("Passed second log")
  }
}

  
let playerName = ""; // Globale variabele

function submitName() {
    const nameInput = document.getElementById('player-name'); // nu correcte id
    playerName = nameInput.value.trim();
  
    if (playerName) {
      console.log("Spelernaam:", playerName);
      alert("Welkom, " + playerName + "!");
    } else {
      alert("Vul aub eerst je naam in!");
    }
  }
  

async function handleSubmit() {
    submitName(); // Welkom berichtje

    await submitFantasyTeam(); // Wacht totdat team is ingezonden

    await loadLeaderboard(); // Refresh het leaderboard
}


async function submitFantasyTeam() {
    const nameInput = document.getElementById('player-name');
    const playerName = nameInput.value.trim();

    if (!playerName) {
      alert("Vul aub je naam in!");
      return;
    }

    const fantasyTeam = {
      player_name: playerName,
      picks: {
        time_value: document.getElementById('time_value').value.trim(),
        coffee: document.getElementById('coffee').value.trim(),
        tires_changed: document.getElementById('tires_changed').value.trim(),
        returns: document.getElementById('returns').value.trim(),
        service_score: document.getElementById('service_score').value.trim(),
        lateness: document.getElementById('lateness').value.trim()
      }
    };

    console.log("Inzenden fantasy team:", fantasyTeam);

    try {
      const response = await fetch('http://127.0.0.1:5000/submit', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(fantasyTeam)
      });

      if (response.ok) {
        alert("✅ Inzending succesvol!");
      } else {
        alert("❌ Fout bij inzenden.");
      }
    } catch (error) {
      console.error('Fout bij verzenden:', error);
      alert("❌ Serverfout. Controleer of Flask server actief is!");
    }
}

async function loadUsers() {
    try {
      const response = await fetch('http://127.0.0.1:5000/employees');
      const employees = await response.json();
  
      const container = document.getElementById('users-list');
      container.innerHTML = '';
  
      employees.forEach(emp => {
        const userHTML = `
          <div class="user-item">
            <div class="user-item-overlay"></div>
            <div class="user-item-content">
              <div class="user-avatar">
                <img src="${emp.photo}" class="avatar-img" alt="Foto van ${emp.name}" />
              </div>
              <div class="user-info">
                <span class="user-overline">Monteur</span>
                <h3 class="user-name">${emp.name}</h3>
              </div>
            </div>
          </div>
        `;
        container.insertAdjacentHTML('beforeend', userHTML);
      });
    } catch (error) {
      console.error("❌ Kon werknemerslijst niet laden:", error);
    }
  }
  
// Start het laden van de werknemers
loadUsers();
loadLeaderboard();  